apps/app/src/app/pages/pages.component.spec.ts

```diff
@@ -31,41 +46,109 @@ describe('PagesComponent', () => {
+    it('should update date every second', fakeAsync(() => {
+      const initialNow = component.now;
+      component.ngOnInit();
+      tick(2000 * 60);
+      discardPeriodicTasks();
+      expect(component.now).toBeGreaterThan(initialNow);
+    }));
   });
 
-  it('should contains a "content-container" class', () => {
-    expect(
-      fixture.nativeElement.querySelector('.content-container')
-    ).toBeTruthy();
-  });
+  describe('track by', () => {
+    it('should return label for navigation link', () => {
+      const label = 'Dashboard';
+      const link = { path: ['/', 'dashboard'], label };
+      expect(component.trackByLabel(0, link)).toBe(label);
+    });
 
-  it('should contains a "content-area" class', () => {
-    expect(fixture.nativeElement.querySelector('.content-area')).toBeTruthy();
-  });
+    it('should return name for language', () => {
+      const lang = { code: LanguageCode.en, name: 'English' } as Language;
+      expect(component.trackByLanguageName(0, lang)).toBe(lang.name);
+    });
 
-  it('should have a router-outlet', () => {
-    const compiled = fixture.debugElement.nativeElement;
-    expect(compiled.querySelector('router-outlet')).toBeTruthy();
+    it('should return application name and version for application', () => {
+      const applicationId = {
+        applicationName: 'Test',
+        applicationVersion: '1.0.0',
+      } as Application['_id'];
+      expect(component.trackByApplicationId(0, applicationId)).toBe(
+        `${applicationId.applicationName}${applicationId.applicationVersion}`
+      );
+    });
   });
 
-  it('should have a header', () => {
-    const compiled = fixture.debugElement.nativeElement;
-    expect(compiled.querySelector('ui-header')).toBeTruthy();
+  describe('language', () => {
+    it('should change language', () => {
+      const lang = { code: LanguageCode.en, name: 'English' } as Language;
+
+      const languageService = TestBed.inject(LanguageService);
+      component.changeLanguage(lang.code);
+
+      expect(languageService.currentLang).toBe(lang.code);
+    });
   });
 
-  it('should contain a title with "Armonik"', () => {
-    const compiled = fixture.debugElement.nativeElement;
-    expect(compiled.querySelector('ui-header .title').textContent).toContain(
-      'ArmoniK'
-    );
+  describe('application', () => {
+    it('should remove application', () => {
+      const application = {
+        applicationName: 'Test',
+        applicationVersion: '1.0.0',
+      } as Application['_id'];
+
+      const settingsService = TestBed.inject(SettingsService);
+      spyOn(settingsService, 'removeCurrentApplication');
+
+      const router = TestBed.inject(Router);
+      spyOn(router, 'navigate');
+
+      component.removeApplication(application);
+
+      expect(settingsService.removeCurrentApplication).toHaveBeenCalledWith(
+        application
+      );
+      expect(router.navigate).toHaveBeenCalledWith(['/', 'dashboard']);
+    });
   });
 
```

Comment:
Tu ne testes pas la fonction isSelected avant de passer à l'ui ?

Reply:
Nan parce que c'est l'ui qui s'en sert pour désactiver ou non le bouton

Reply:
Du coup tu teste cette fonction au travers des tests sur l'ui ?

Reply:
J'ai ajouté un test à ce sujet `should have button from current lang disabled`